core/src/test/java/org/elasticsearch/index/query/PercolatorQueryBuilderTests.java

```diff
@@ -75,6 +79,16 @@ private PercolatorQueryBuilder doCreateTestQueryBuilder(boolean indexedDocument)
         }
     }
 
+    /**
```

Comment:
I don't understand why comparing `BytesReference` would fail if the content is the same. Regardless of the location of `document` field, it is just parsing its content so it should always be the same? Or am I missing something here?